src/megaapi_impl.cpp

```diff
@@ -6420,6 +6439,35 @@ bool MegaApiImpl::createLocalFolder(const char *path)
     return success;
 }
 
+MegaErrorPrivate MegaApiImpl::createLocalFolder(LocalPath & localPath)
+{
+    SdkMutexGuard g(sdkMutex);
+    auto da = client->fsaccess->newfileaccess();
+    if (!da->fopen(localPath, true, false))
+    {
+        if (!client->fsaccess->mkdirlocal(localPath))
+        {
+            da.reset();
```

Comment:
no need for these 4 calls to da.reset(), as da's destructor will always be called before exiting the function.  This is a case where we can rely on RAII to keep our code small and tidy.   thanks